include/seqan3/io/exception.hpp

```diff
@@ -74,4 +74,12 @@ struct parse_error : std::runtime_error
     {}
 };
 
+//!\brief Thrown if there there was an unexpected end of input, e.g. the end-of-file marker was reached unexpectedly.
+struct unexpected_end_of_error : parse_error
```

Comment:
this really sounds weird, maybe `unexpected_end_of_file` (do we need error in it?) or `unexpected_eof_error` would be better?